### PR TITLE
Add Swedish Forest Damage dataset

### PR DESCRIPTION
**Pull request by nilsleh**
This PR adds the [Forest Damage](https://lila.science/datasets/forest-damages-larch-casebearer/) dataset for tree detection and tree health classification based on drone imagery coming in `.jpg` images. 

Features:
- 1543 images with 101,878 tree bounding boxes
- subset of 840 images contains 44,522 tree health annotations

Example:
Screenshot from 2022-03-13 15-14-15

### CODE REVIEW

File: tests/datasets/test_forestdamage.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+
+def download_url(url: str, root: str, *args: str) -> None:
+    shutil.copy(url, root)
+
+
+class TestForestDamage:
+    @pytest.fixture
+    def dataset(
+        self, monkeypatch: Generator[MonkeyPatch, None, None], tmp_path: Path
```
Comment (calebrob6) [inline tests/datasets/test_forestdamage.py:26]: This "Generator" type isn't needed anymore  (see some of the other tests)

File: tests/datasets/test_forestdamage.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    ) -> ForestDamage:
+        monkeypatch.setattr(  # type: ignore[attr-defined]
```
Comment (calebrob6) [inline tests/datasets/test_forestdamage.py:28]: These ignore attr-defined won't be needed either

File: torchgeo/datasets/forestdamage.py [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+import numpy as np
+import torch
+from PIL import Image
+from torch import Tensor
+
+from .geo import VisionDataset
+from .utils import check_integrity, download_and_extract_archive, extract_archive
+
+
+def parse_pascal_voc(path: str) -> Dict[str, Any]:
```
Comment (calebrob6) [inline torchgeo/datasets/forestdamage.py:22]: This seems like something to put in `torchgeo.datasets.utils`

Comment (nilsleh) [reply torchgeo/datasets/forestdamage.py:22]: In the FAIR1M dataset, annotations also come in PASCAL VOC format, but the xml files differ so, like in the FAIR1M dataset, I put the parsing function in `forestdamage.py` itself.

File: torchgeo/datasets/forestdamage.py [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+    * other (0)
+    * healthy (1)
+    * light damage (2)
+    * high damage (3)
+
+    If the download fails or stalls, it is recommended to try azcopy
+    as suggested `here <https://lila.science/faq>`__. It is expected that the
+    downloaded data file with name `Data_Set_Larch_Casebearer`
+    can be found in `root`.
```
Comment (adamjstewart) [inline torchgeo/datasets/forestdamage.py:81]: These will be in italics, not monotype. Is that what you want?

**Comment by adamjstewart** [comment]
To fix the failing tests just run:
```console
$ pip install pyupgrade
$ pyupgrade --py37-plus tests/data/forestdamage/data.py
```

File: tests/data/forestdamage/data.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import shutil
+import xml.etree.cElementTree as ET
+
+import numpy as np
+from PIL import Image
+
+SIZE = 32
+
+np.random.seed(0)
+random.seed(0)
```
[the remark this answers is not in the document]
Comment (nilsleh) [reply tests/data/forestdamage/data.py:13]: It was done in `data.py` for DFC2022, so I have done it for all the data scripts as well.

**Review by calebrob6 — DISMISSED** [review]

File: tests/data/forestdamage/data.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (adamjstewart) [reply tests/data/forestdamage/data.py:13]: I don't like to blindly copy and paste things from one file to the next. We should only set `random.seed` if we actually use `random` elsewhere to generate random numbers.

File: torchgeo/datasets/forestdamage.py [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+            raise RuntimeError(
+                "Dataset not found in `root` directory, either specify a different"
+                + " `root` directory or manually download "
+                + "the dataset to this directory."
+            )
+
+        # else download the dataset
+        self._download()
+
+    def _download(self) -> None:
```
[the remark this answers is not in the document]
Comment (nilsleh) [reply torchgeo/datasets/forestdamage.py:243]: That is what I intended with the step above line 225-230, where if the downloaded zip file is found, it will only be extracted. Or would you prefer that in the case no zip file is found, the download and extract also happen in two different steps?

[the remark this answers is not in the document]
Comment (adamjstewart) [reply torchgeo/datasets/forestdamage.py:243]: Ah, I guess this is fine then. I usually have separate `_download` and `_extract` functions but in this case it's just a single file.

**Review by adamjstewart — APPROVED** [review]
